### PR TITLE
Updates die to use the roll() proc + initialize()

### PR DESCRIPTION
**Pull request by praisenarsie**
plus some tweaks

credits to somepotato for enlightening us with the roll() proc on #coderbus

### CODE REVIEW

File: code/game/objects/items/dice.dm [hunk 1]
```diff
@@ -131,6 +131,7 @@
 	name = "d100"
 	desc = "A die with one hundred sides! Probably not fairly weighted..."
 	icon_state = "d100"
+	w_class = WEIGHT_CLASS_SMALL
```
Comment (coiax) [inline code/game/objects/items/dice.dm]: declare changes.

Comment (praisenarsie) [reply code/game/objects/items/dice.dm]: > plus some tweaks

**Comment by Kromgar** [comment]
Does this include the die of fate from lavaland? OR maybe it's just a subtype of normal dice?

**Comment by praisenarsie** [comment]
it should

**Pull request merged**